spec/controllers/jira/status/push_controller_spec.rb

```diff
@@ -44,14 +44,70 @@
       exception = ArgumentError.new("Bad args")
       expect(DeployMailer).to receive(:deployment_email) { raise exception }
 
-      expect { get :deploy_email, id: '12345678' }.to raise_error(exception)
+      expect { get :deploy_email, id: '12345678', service_name: 'rs_west' }.to raise_error(exception)
       expect(@push.reload.email_sent).to eq(false)
     end
 
     it 'redirects to summary page if not commit not found with sha matchin id' do
-      get :deploy_email, id: 'abc'
+      get :deploy_email, id: 'abc', service_name: 'rs_west'
       expect(response).to have_http_status(302)
       expect(response.redirect_url).to eq('http://test.host/400')
     end
   end
+
+  describe 'view routes' do
+    let(:web_ref) { AncestorRef.find_or_create_by!(service_name: 'web') }
+    let(:web_push) { Push.create!(status: :success, branch: @branch, head_commit: @commit, ancestor_ref: web_ref) }
+
+    describe 'branch' do
+      subject { get :branch, branch: 'feature_branch' }
+
+      it 'renders edit page for web ancestor ref' do
+        dbl = double("ActiveRecord Relation", first!: @branch, for_ancestor: [web_push])
+        expect(Branch).to receive(:where).and_return(dbl)
+        expect(@branch).to receive(:pushes).and_return(dbl)
+        expect(dbl).to receive(:for_ancestor).with("web").and_return([web_push])
+
+        expect(subject).to render_template("jira/status/push/edit")
+      end
+    end
+
+    describe 'summary' do
+      subject { get :summary }
+
+      it 'renders summary page for web ancestor ref with master branch' do
+        dbl = double("ActiveRecord Relation", first!: @branch, for_ancestor: [web_push])
+        expect(Branch).to receive(:where).with(name: "master").and_return(dbl)
+        expect(@branch).to receive(:pushes).and_return(dbl)
+        expect(dbl).to receive(:for_ancestor).with("web").and_return([web_push])
+
+        expect(subject).to render_template("jira/status/push/summary")
+      end
+    end
+
+    # TODO: improve test coverage for this action
```

Comment:
Was this something you were expecting to address in this PR or in the future?

Reply:
In the future. This action had zero tests, so I added some for the functionality in this PR with a go right case. I added test coverage for other actions too, but this one has a lot happening and unpacking all of that to write full test coverage seemed out of scope

Reply:
Ack, ok 👍  I'd suggest just updating the comment to indicate who it came from so that anyone in the future will more easily know where to go if they have questions (plus the comment keeps ownership even if someone moves this line around)

E.g. `TODO - LVieira/Phoenix: ...` (in most cases just `Phoenix` would be ok but in this case since no team owns pre_deploy_checker having your name there as well could be helpful)